public/app/core/components/TransformersUI/FilterByValueTransformerEditor.tsx

```diff
@@ -0,0 +1,231 @@
+import React, { useMemo } from 'react';
+import {
+  DataTransformerID,
+  standardTransformers,
+  TransformerRegistyItem,
+  TransformerUIProps,
+  getFieldDisplayName,
+  Field,
+  DataFrame,
+} from '@grafana/data';
+import { getAllFieldNamesFromDataFrames } from './OrganizeFieldsTransformerEditor';
+import { Select, Button, Input } from '@grafana/ui';
+
+import {
+  FilterByValueTransformerOptions,
+  ValueFilter,
+} from '@grafana/data/src/transformations/transformers/filterByValue';
+
+import { valueFiltersRegistry, ValueFilterID } from '@grafana/data/src/transformations/valueFilters';
+
+function FilterSelectorRow(props: any) {
+  const { fieldNameOptions, onDelete, onConfigChange, config, fieldType } = props;
+
+  console.log('props', props);
+
+  // Find filter types that fit the chosen field type
+  let filterTypeOptions = valueFiltersRegistry
+    .list()
+    .filter(element => {
+      if (element.supportedFieldTypes) {
+        return !element.supportedFieldTypes ? true : element.supportedFieldTypes.includes(fieldType);
+      } else {
+        return true;
+      }
+    })
+    .map(item => ({ value: item.id, label: item.name, description: item.description }));
+
+  let filterInfo = valueFiltersRegistry.get(config.filterType);
+  let filterValid = filterInfo.getInstance({
+    filterExpression: config.filterExpression,
+    fieldType: fieldType,
+  }).isValid;
+
+  let fieldNameInvalid =
+    config.fieldName === null ||
+    fieldNameOptions.filter((item: Record<string, any>) => item.value === config.fieldName).length === 0;
+  let filterTypeInvalid =
+    !fieldNameInvalid && filterInfo.supportedFieldTypes && !filterInfo.supportedFieldTypes.includes(fieldType);
+  let filterExpressionInvalid = !fieldNameInvalid && !filterTypeInvalid && !filterValid;
```

Comment:
```suggestion
  const filterExpressionInvalid = !fieldNameInvalid && !filterTypeInvalid && !filterValid;
```

I would suggest using `const` instead of `let` on all the places where the value of the variable never change.